### ISSUE TITLE
New admin2 master console not works - Failed to load resource

### ISSUE THREAD

**Issue by breedjaram32** [keycloak/keycloak#14135]
### Describe the bug

After upgrading from 18.0.1 to 19.0.1 is a problem with default admin2 security console. 
After logging in, the outline of the administration flashes briefly and after see white page only.

The cause is most likely in `ENV KC_HTTP_RELATIVE_PATH=/auth`

Report from browser console
<img width="1439" alt="image" src="https://user-images.githubusercontent.com/77382127/187871752-4788d767-05b3-472c-83ea-88e9f4545924.png">

### Version

19.0.1

### How to Reproduce?

**My Dockerfile:**

```
FROM quay.io/keycloak/keycloak:19.0.1 as builder

ENV KC_METRICS_ENABLED=true
ENV KC_HEALTH_ENABLED=true
ENV KC_HTTP_RELATIVE_PATH=/auth
ENV KC_CACHE=ispn
ENV KC_CACHE_STACK=kubernetes
ENV KC_DB=mariadb


RUN /opt/keycloak/bin/kc.sh build 

FROM quay.io/keycloak/keycloak:19.0.1
COPY --from=builder /opt/keycloak/ /opt/keycloak/

WORKDIR /opt/keycloak
ENV KEYCLOAK_ADMIN=admin
ENV KEYCLOAK_ADMIN_PASSWORD=PW

ENV KC_DB_URL=jdbc:mariadb://URL:3306/keycloak
ENV KC_DB_USERNAME=DATABASE
ENV KC_DB_PASSWORD=PW
ENV KC_HOSTNAME=HOSTNAME_URL
ENV KC_HOSTNAME_ADMIN=HOSTNAME_URL_ADMIN
ENV KC_PROXY=edge

ENV KC_HOSTNAME_STRICT_BACKCHANNEL=true
ENTRYPOINT ["/opt/keycloak/bin/kc.sh", "start", "--optimized"]
```

**Comment by lexcao** [comment]
Hi, Could you please try to add `ENV KC_HTTP_RELATIVE_PATH=/auth` to the start command of docker?


See more details from #14048

**Comment by breedjaram32** [comment]
Hi @lexcao 
I found this issue too https://github.com/keycloak/keycloak/issues/14048 yesterday but not works for me. Same problem with

```
RUN /opt/keycloak/bin/kc.sh build
FROM quay.io/keycloak/keycloak:19.0.1
COPY --from=builder /opt/keycloak/ /opt/keycloak/
ENTRYPOINT ["/opt/keycloak/bin/kc.sh", "start", "--optimized"]
```

but I`am using "`start", "--optimized"`  in ENTRYPOINT. You without --optimized but without --optimized Keycloak not works for me.

**Comment by lexcao** [comment]
Are you trying this?
```
RUN /opt/keycloak/bin/kc.sh build
FROM quay.io/keycloak/keycloak:19.0.1
COPY --from=builder /opt/keycloak/ /opt/keycloak/

ENV KC_HTTP_RELATIVE_PATH=/auth    <------ add env here

ENTRYPOINT ["/opt/keycloak/bin/kc.sh", "start", "--optimized"]
```

**Comment by breedjaram32** [comment]
Yes I tried this.
In my last comment have a mistake. I missing Workdir and ENV (but i tried with them)
Still not works.

```
RUN /opt/keycloak/bin/kc.sh build
FROM quay.io/keycloak/keycloak:19.0.1
COPY --from=builder /opt/keycloak/ /opt/keycloak/
WORKDIR /opt/keycloak
ENV KC_HTTP_RELATIVE_PATH=/auth
ENTRYPOINT ["/opt/keycloak/bin/kc.sh", "start", "--optimized"]
```

**Comment by luchsamapparat** [comment]
We seem to have a similar problem.

Our Dockerfile looks like this (we're using the latest nightly build here):

```
FROM quay.io/keycloak/keycloak:nightly as builder

ENV KC_HTTP_RELATIVE_PATH=/auth

# ...

RUN /opt/keycloak/bin/kc.sh build

FROM quay.io/keycloak/keycloak:nightly
COPY --from=builder /opt/keycloak/ /opt/keycloak/

WORKDIR /opt/keycloak

# ...

ENV KC_PROXY=edge
ENV KC_HTTP_ENABLED=true
ENV KC_HOSTNAME_STRICT_BACKCHANNEL=true

ENTRYPOINT ["/opt/keycloak/bin/kc.sh"]
CMD ["start", "--optimized"]
```

The container is configured to run with these environment variables:

```
KC_HOSTNAME_URL="https://[domain]portal.team-beverage.dev/login/auth"
KC_HOSTNAME_ADMIN_URL="https://[domain]/login/admin/im/auth"
```

(`KC_HOSTNAME_ADMIN_URL` is currently, as of 19.0.1, only available in nightly builds)

So in summary,

- we prefix _all_ requests with `/login` (via our proxy)
- we set the context path for Keycloak to `/auth` (this also applies for _all_ requests)
- we additionally prefix all requests for the Administration with `/admin/im`

We try to achieve this:

Client Authentication uses URLs starting with `https://[domain]/login/auth/[...]`
The Administration Console uses URLs starting with `https://[domain]portal.team-beverage.dev/login/admin/im/auth/[...]`

(the reasoning behind this is not relevant, but there _is_ a reason)

Our proxy is configured accordingly.

*What works?*
- I can authenticate using `keycloak-js` (via `/login/auth/`)
- I can open the "Welcome to Keycloak" page (`/login/admin/im/auth/`)
- I can open the Administration Console Login (`/login/auth/realms/master/protocol/openid-connect/auth?...`)
- I can successfully login using the Administration Console Login

*What does not work?* 
The Administration Console doesn't render because these some requests fail with an `404` error:

![image](https://user-images.githubusercontent.com/875017/189917013-c15bd86e-85bb-4c1e-9444-6aa184286938.png)

As you can see, all four requests are clearly related to the admin console. Therefore, I would expect that their paths start with `/login/admin/im/auth/` (as defined by `KC_HOSTNAME_ADMIN_URL`). Instead, they start with `/login/auth/admin`, probably because of the `KC_HOSTNAME_URL`.

*Why should this work?*
`KC_HOSTNAME_ADMIN_URL` is defined as "the base URL for accessing the administration console" ([see here](https://github.com/keycloak/keycloak/blob/52aad0bbdc409e2b60271d543ff218300fc726d5/quarkus/config-api/src/main/java/org/keycloak/config/HostnameOptions.java#L22)). Assets that are clearly related to the admin console should therefore adhere to this setting.

**Comment by stianst** [comment]
We believe this is fixed in https://github.com/keycloak/keycloak-ui/pull/3373

Could you please try the nightly release (quay.io/keycloak/keycloak:nightly) and check if that works?

**Comment by breedjaram32** [comment]
Hi @stianst great! Admin2 works perfect with nightly build.

**Comment by stianst** [comment]
@breedjaram32 awesome thanks, we'll get this included in 19.0.3, which should be out early next week.
